packages/next/client/dev/error-overlay/hot-dev-client.js

```diff
@@ -98,11 +99,7 @@ function handleSuccess() {
 
   // Attempt to apply hot updates or reload.
   if (isHotUpdate) {
-    tryApplyUpdates(function onSuccessfulHotUpdate(hasUpdates) {
-      // Only dismiss it when we're sure it's a hot update.
-      // Otherwise it would flicker right before the reload.
-      onFastRefresh(hasUpdates)
-    })
+    tryApplyUpdates(onBeforeFastRefresh, onFastRefresh)
```

Comment:
I wasn't sure whether to just inline these function calls directly within `tryApplyUpdates`, more on that in a comment below.